dmci/api/worker.py

```diff
@@ -104,48 +100,60 @@ def validate(self, data):
             if not valid:
                 return valid, msg, data
 
+            if (".dev" in self._namespace and self._conf.env_string != "dev") or (
+                ".staging" in self._namespace and self._conf.env_string != "staging"
+            ):
+                msg = (
+                    f"Namespace {self._namespace} does not match "
+                    f"the env {self._conf.env_string}"
+                )
+                return False, msg, data
+
```

Comment:
Do I understand correctly that this is the main change? The rest seems to be syntax..?

Reply:
yes